consensus/spos/bls/subroundBlock.go

```diff
@@ -384,7 +384,8 @@ func (sr *subroundBlock) receivedBlockBodyAndHeader(ctx context.Context, cnsDta
 	sr.Body = sr.BlockProcessor().DecodeBlockBody(cnsDta.Body)
 	sr.Header = sr.BlockProcessor().DecodeBlockHeader(cnsDta.Header)
 
-	if sr.Data == nil || check.IfNil(sr.Body) || check.IfNil(sr.Header) {
+	isInvalidData := check.IfNil(sr.Body) || sr.isInvalidHeaderOrData()
```

Comment:
Maybe just isInvalidHeader() and keep the sr.Data check here?

Reply:
sr.Data is also checked on the other occurrence on L480

Reply:
Ok, it is a little strange to separate body check and header check and to group it this way: 1. checks for body and 2. checks for data and header. As data is a separate filed in a DTO, it is not contained or related to body or header But I think it could remain like this.

Reply:
So the sr.Data is related to the Body?

Reply:
Actually it is not related to Body nor Header, but it is the id of the current round (hash of the proposed header, without some fields set, like: signature, leader siganture, pubkeybitmap), used and verified in each subround. This sr.Data is the id verified first for each message received on consensus topic.